### ISSUE TITLE
can't preserve aspect ratio when screen size shrinks or grows?

### ISSUE THREAD

**Issue by dmccallie** [DougAnderson444/svelte-zoom-action#2]
Me again,

I'm trying to create a page where images display in their natural aspect ratio, and respond appropriately to screen size growing and shrinking (meaning that aspect ratio stays correct. I've been using "max-width: 100%" to make this work.

I'd like to add `use:pzoom` to these images, so that someone could zoom in for details, even on a smaller screen.  The best I can get so far is for the width to shrink appropriately, but the height does not, and thus the image gets squeezed.  I've tried all sorts of hacks, but I probably am either asking for the impossible, or revealing how little I understand about HTML/CSS and resizing.

Here's a simple example that hopefully demonstrates the question. 

Am I missing the right way to do this, or is this impossible?

Thanks!

```svelte
<script>
import { pzoom } from '@douganderson444/panzoom-node'
let zoomable, container;    
</script>


<!-- without zoomable, this works to shrink/grow image in proper aspect -->
<div>
    <img src= "https://photos.smugmug.com/---.jpg"
    alt=""
    class="test"
    >
</div>

<!-- this one zooms/pans but but shrinking does not preserve aspect ratio  -->
<div class="container" bind:this={container}>
    <img src= "https://photos.smugmug.com/---.jpg"
    alt=""
    bind:this={zoomable}
    use:pzoom={{ panAnywhere: true }}
    >
</div>

<style>
    .test {
       max-width: 100% 
    }

   .container {
	border: 3px solid red;
	height: 400px;  /* wont work without something here! */
	width: 600px;
        max-width: 100%;
    }
</style>

```

**Comment by DougAnderson444** [comment]
It's useful to include a reproduction ("repro") when you run into an issue, like this:

https://svelte.dev/repl/63cfec80e79a458cbe6dffbd9feb4ee8?version=3.55.1

Just wrap your image in a zoomable container. 

```svelte
<script>
import { pzoom } from '@douganderson444/panzoom-node@1.2.2'
</script>

<h1>
  Here kitty
</h1>

<div use:pzoom={{ panAnywhere: true }}>
  <img src='https://i.imgur.com/0uBcgbm.jpeg' alt='kitty' />
</div>
```


I suppose saying that *any* HTML Node can be zoomable was misleading, doesn't seem to work on the img element directly. But wrapping it works, keeps the aspect ratio as you'd expect.

**Comment by dmccallie** [comment]
Thanks for the helpful reply and example. (It's good to learn about the repl.)

In my case I also need a container for the zoomable that wraps the image, since I don't want the image bound to the page body.

My goal is to have the image load into a pre-defined container width, and to have the image sized such that the width of the image magically expands to the width of the container, and the height grows as much as needed to maintain the aspect ratio (I here assume that the native width is greater than the height.)

With a regular (non-zoom) image, I can use `flex display`, and the `max-width:100%` setting on the outer container and this works fine.

But if I use pzoom, I can't figure out how to get the resulting image to show at all unless I pre-define both the `height` and `width`.

In other words, as best I can tell, I have to specify height and width of the zoom-wrapping container _before_ the image is loaded.  Is that correct?  Is there any way to make this dynamic?

Here's an attempt at a REPL.

https://svelte.dev/repl/f362d52bc9fa4ef5b37e47130cfd59ed?version=3.55.1

Thanks again.

**Comment by DougAnderson444** [comment]
Hi,

Did you try setting the image width to 100%?

 <img **width="100%"** src='https://i.imgur.com/0uBcgbm.jpeg' alt='zoom-kitty' />?

**Comment by DougAnderson444** [comment]
Works with 100% img width, see:

https://svelte.dev/repl/7d6781a7f7554ece8d2ba214f3d42367?version=3.55.1

**Comment by dmccallie** [comment]
Thank you. Adding `<img width=100%/>`  certainly helps.

I think there is still a limitation compared to how a regular image resizes when placed inside a `<flexbox>` with `flex-direction: column` in that the non-pzoom image automatically finds the size of the column width, and then changes width dynamically while maintaining both full width and aspect-accurate height, whereas the augmented `pzoom` image appears to require that a specific height and width be specified up front.

I have not been able to find a way for the pzoom image to automatically pick up the flexbox column width. (Perhaps reflecting my lack of deep HTML/CSS knowledge!) I tried `width: auto`, and `width: inherit`  and `height:100%` and other combinations using `flex-grow` and `flex-shrink`  but always had to set an image-specific height and/or width to get it to work. The closest I got was setting the height to a very large number, which works except that the excess height pushes the next element far down the page. Setting `overflow:hidden` did not fix it.

Maybe there is some way to dynamically grab the current flexbox column width, and then programmatically set the width of the zoomable image before it is displayed? Maybe something using onMount()?

However, I think I can work around this by redefining my app's flow, so it's not a problem (so far...)

Thanks again for all your help.
